lib/promenade/karafka/consumer_subscriber.rb

```diff
@@ -0,0 +1,45 @@
+require "promenade/karafka/subscriber"
+
+module Promenade
+  module Karafka
+    class ConsumerSubscriber < Subscriber
+      attach_to "consumer.karafka"
+
+      Promenade.histogram :kafka_consumer_batch_processing_duration do
+        doc "Consumer message processing latency in seconds"
+        buckets :network
+      end
+
+      Promenade.counter :kafka_consumer_messages_processed do
+        doc "Messages processed by this consumer"
+      end
+
+      def consumed(event)
+        consumer = event.payload[:caller]
+        messages = consumer.messages
+        batch_processing_duration = event.payload[:time] / 1000
```

Comment:
I can't find the place in the karafka source where this event is emmited so I can't double check.

But if the units are microseconds then this needs to be:

```suggestion
        batch_processing_duration = event.payload[:time] / 1000000.0
```

📝 note also that you need to devide by a float here or you essentially ignore most of the buckets in the histogram

Reply:
The units are currently in miliseconds 👍 

Will update to divide by a float, thanks!